docs/manual/scala/guide/build/ScalaComponents.md

```diff
@@ -0,0 +1,96 @@
+# Scala Components
+
+The Lagom scaladsl requires building an Application that mixes in all the `Components` that will enable the required features for your code. This means each `Application` will need a specific set of cake slices. Not only that but `Dev` mode and `Prod` mode need adjusting since the environments where the Service is run vary.
+
+Imagine you have a `HelloService` and it's implementation and you are only missing the Appliciation. You could prepare:
+
+@[lagom-application](code/ServiceImplementation.scala)
+
+That is a simple enough Application but it's not complete for running. It extends [`LagomApplication`](api/com/lightbend/lagom/scaladsl/server/LagomApplication.html) so it has all the basic bits and pieces but you need to specify at least a [[Service Locator|ServiceLocator]] for each environment Lagom supports:
+
+@[lagom-loader](code/ServiceImplementation.scala)
+
+Once the `Application` was ready the `ApplicationLoader` is responsible to instantiate and provide the final bits of the cake. See how the `ApplicationLoader` defines two methods: `loadDevMode()` and `load()` so you can mix-in different ServiceLocators.
+
+This is a brief introduction to Lagom Application cake, there's more information in [[Wiring together a Lagom application|ServiceImplementation#Wiring-together-a-Lagom-application]] section.
+
+## Components
```

Comment:
You just confirmed I suspicion I had... I spent some time grouping them but when everything was done I felt a little separation/categorization would be helpful. Will do that.

Also, I don't like that all Components seem to have the sam importance. You'll rarely use a `NoServiceLocator` and you'll often use an `AhcWSComponents`. I want to make it clear that ordering is unrelated to importance.